ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/ReadsKeySets.hs

```diff
@@ -202,3 +205,34 @@ forwardTableKeySets ::
             (LedgerTables l ValuesMK)
 forwardTableKeySets dblog =
   forwardTableKeySets' (changelogDiffAnchor dblog) (changelogDiffs dblog)
+
+{-------------------------------------------------------------------------------
+  Statistics
+-------------------------------------------------------------------------------}
+
+-- | Get the number of entries that are in a ledger table associated with the
+-- current ledger state.
+getCurrentLedgerTableSize ::
+     (Monad m, HasLedgerTables l)
+  => LedgerDB l
+  -> LedgerBackingStore m l
+  -> m (Either (WithOrigin SlotNo, WithOrigin SlotNo) Int)
+getCurrentLedgerTableSize lgrDb lbs = do
+    Statistics{sequenceNumber = seqNo', numEntries = n} <- bsStat bs
+    if seqNo /= seqNo' then
+      pure $ Left (seqNo, seqNo')
```

Comment:
This probably should be wrapped in some error type, it is always difficult to see which one is which in this tuples when you get an error